tiledb/sm/config/config.h

```diff
@@ -706,6 +711,10 @@ class Config {
   /** Compares configs for equality. */
   bool operator==(const Config& rhs) const;
 
+  /** Get all config params taking into account environment variables */
+  const std::map<std::string, std::string> get_all_params_from_config_or_env()
+      const;
+
```

Comment:
Maybe a better way would be to populate the config options with the environment variables at the time the `Config` gets created, but it might introduce complications when unsetting values.